h264/bits/bitreader.go

```diff
@@ -0,0 +1,119 @@
+/*
+DESCRIPTION
+  bitreader.go provides a bit reader implementation that can read or peek from
+  an io.Reader data source.
+
+AUTHORS
+  Saxon Nelson-Milton <saxon@ausocean.org>, The Australian Ocean Laboratory (AusOcean)
+
+LICENSE
+
+  Copyright (c) 2009 The Go Authors. All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+    * Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above
+  copyright notice, this list of conditions and the following disclaimer
+  in the documentation and/or other materials provided with the
+  distribution.
+    * Neither the name of Google Inc. nor the names of its
+  contributors may be used to endorse or promote products derived from
+  this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+// Package bits provides a bit reader implementation that can read or peek from
+// an io.Reader data source.
+package bits
+
+import (
+	"bufio"
+	"io"
+)
+
+// BitReader is a bit reader that provides methods for reading bits from an
+// io.Reader source.
+type BitReader struct {
+	r    *bufio.Reader
+	n    uint64
+	bits uint
+}
+
+// NewBitReader returns a new BitReader.
+func NewBitReader(r io.Reader) *BitReader {
+	byter, ok := r.(*bufio.Reader)
+	if !ok {
+		byter = bufio.NewReader(r)
+	}
+	return &BitReader{r: byter}
+}
+
+// ReadBits reads n bits from the source and returns as an uint64.
```

Comment:
Make the comment match the signature; `bits`/`n`. I would use `n` Also I'd lend from the bzip2 doc comments which arguable clearer.